docs/docs/flink-maintenance.md

```diff
@@ -0,0 +1,393 @@
+
+### Supported Features (Flink)
+
+#### ExpireSnapshots
+Removes old snapshots and their files. Internally uses `cleanExpiredFiles(true)` when committing, so expired metadata/files are cleaned up automatically.
+
+```java
+.add(ExpireSnapshots.builder()
+    .maxSnapshotAge(Duration.ofDays(7))
+    .retainLast(10)
+    .deleteBatchSize(1000))
+```
+
+#### RewriteDataFiles
+Compacts small files to optimize file sizes. Supports partial progress commits and limiting maximum rewritten bytes per run.
+
+```java
+.add(RewriteDataFiles.builder()
+    .targetFileSizeBytes(256 * 1024 * 1024)
+    .minFileSizeBytes(32 * 1024 * 1024)
+    .partialProgressEnabled(true)
+    .partialProgressMaxCommits(5))
+```
+
+### Lock Management
+
+The `TriggerLockFactory` is essential for coordinating maintenance tasks. It prevents concurrent maintenance operations on the same table, which could lead to conflicts or data corruption. This locking mechanism is necessary even for a single job, as multiple instances of the same task could otherwise conflict.
+
+#### Why Locks Are Needed
+- **Concurrent Access**: Multiple Flink jobs may attempt maintenance simultaneously
+- **Data Consistency**: Ensures only one maintenance operation runs per table at a time
+- **Resource Management**: Prevents resource conflicts and scheduling issues
+- **Avoid Duplicate Work**: Even when only a single compaction job is scheduled, multiple instances could attempt the same operation, leading to redundant work and wasted resources.
+
+#### Supported Lock Types
+
+##### JDBC Lock Factory
+Uses a database table to manage distributed locks:
+
+```java
+Map<String, String> jdbcProps = new HashMap<>();
+jdbcProps.put("jdbc.user", "flink");
+jdbcProps.put("jdbc.password", "flinkpw");
+jdbcProps.put("flink-maintenance.lock.jdbc.init-lock-tables", "true"); // Auto-create lock table if it doesn't exist
+
+TriggerLockFactory lockFactory = new JdbcLockFactory(
+    "jdbc:postgresql://localhost:5432/iceberg", // JDBC URL
+    "catalog.db.table",                         // Lock ID (unique identifier)
+    jdbcProps                                   // JDBC connection properties
+);
+```
+
+##### ZooKeeper Lock Factory
+Uses Apache ZooKeeper for distributed locks:
+
+```java
+TriggerLockFactory lockFactory = new ZkLockFactory(
+    "localhost:2181",       // ZooKeeper connection string
+    "catalog.db.table",     // Lock ID (unique identifier)
+    60000,                  // sessionTimeoutMs
+    15000,                  // connectionTimeoutMs
+    3000,                   // baseSleepTimeMs
+    3                       // maxRetries
+);
+```
+
+### Quick Start
+
+The following example demonstrates the implementation of automated maintenance for an Iceberg table within a Flink environment.
+
+```java
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+TableLoader tableLoader = TableLoader.fromHadoopTable("path/to/table");
```

Comment:
`fromHadoopTable` -> Can we change this to `hive` or something else?
We would like to remove the references of the HadoopTable from the docs as it is not encouraged in production environments